packages/reference/src/common/MultipleReferenceEditor.tsx

```diff
@@ -21,106 +24,66 @@ type EditorProps = ReferenceEditorProps &
     children: (props: ReferenceEditorProps & ChildProps) => React.ReactElement;
   };
 
-type EditorState = {
-  canCreateEntity: boolean;
-};
-
-class Editor extends React.Component<EditorProps, EditorState> {
-  constructor(props: EditorProps) {
-    super(props);
-    this.state = {
-      canCreateEntity: true,
-    };
-  }
-
-  componentDidMount() {
-    if (this.props.entityType === 'Asset') {
-      this.props.sdk.access.can('create', 'Asset').then((value) => {
-        this.setState({ canCreateEntity: value });
-      });
-    }
-  }
-
-  canCreateEntity = () => {
-    if (this.props.parameters.instance.showCreateEntityAction === false) {
-      return false;
-    }
-    return this.state.canCreateEntity;
-  };
-
-  canLinkEntity = () => {
-    if (this.props.parameters.instance.showLinkEntityAction !== undefined) {
-      return this.props.parameters.instance.showLinkEntityAction;
-    }
-    return true;
-  };
-
-  onSortStart: SortStartHandler = (_, event) => event.preventDefault();
-
-  onSortEnd: SortEndHandler = ({ oldIndex, newIndex }) => {
-    const newItems = arrayMove(this.props.items, oldIndex, newIndex);
-    this.props.setValue(newItems);
-  };
+function Editor(props: EditorProps) {
+  const { items, setValue, entityType } = props;
+  const { canCreateEntity, canLinkEntity } = useEntityPermissions(props);
+
+  const onSortStart: SortStartHandler = useCallback((_, event) => event.preventDefault(), []);
+  const onSortEnd: SortEndHandler = useCallback(
+    ({ oldIndex, newIndex }) => {
+      const newItems = arrayMove(items, oldIndex, newIndex);
+      setValue(newItems);
+    },
+    [items, setValue]
+  );
 
-  onCreate = (id: string) => {
-    this.props.setValue([
-      ...this.props.items,
-      {
-        sys: {
-          type: 'Link',
-          linkType: this.props.entityType,
-          id,
-        },
-      },
-    ]);
-  };
+  const onCreate = useCallback(
+    (id: string) => {
+      setValue([...items, { sys: { type: 'Link', linkType: entityType, id } }]);
+    },
+    [setValue, items, entityType]
+  );
 
-  onLink = (ids: string[]) => {
-    this.props.setValue([
-      ...this.props.items,
-      ...ids.map((id) => {
-        return {
-          sys: {
-            type: 'Link',
-            linkType: this.props.entityType,
-            id,
-          } as const,
-        };
-      }),
-    ]);
-  };
+  const onLink = useCallback(
+    (ids: string[]) => {
+      setValue([
+        ...items,
+        ...ids.map((id) => ({ sys: { type: 'Link', linkType: entityType, id } as const })),
+      ]);
+    },
+    [setValue, items, entityType]
+  );
 
-  render() {
-    const validations = fromFieldValidations([
-      ...this.props.sdk.field.validations,
-      ...(this.props.sdk.field.items?.validations ?? []),
-    ]);
+  const validations = fromFieldValidations([
+    ...props.sdk.field.validations,
+    ...(props.sdk.field.items?.validations ?? []),
+  ]);
+  const linkActionsProps = useLinkActionsProps({
```

Comment:
Seems a bit expensive to do this for every link. Can't we just do it once and then reuse the same set of props on all links and for `renderCustomActions`?

Reply:
Actually I think that's how it works right now. `Editor` is rendered just once and then renders the children - items. So `useLinkActionsProps` is also supposed to be called once (and it also memoized the returned value). Then these props are passed to the custom card renderer and to `LinkEntityActions`.